src/specialitensornetworks.jl

```diff
@@ -21,49 +21,62 @@ end
 """
 BUILD Z OF CLASSICAL ISING MODEL ON A GIVEN GRAPH AT INVERSE TEMP BETA
 H = -\\sum_{(v,v') \\in edges}\\sigma^{z}_{v}\\sigma^{z}_{v'}
-TAKE AS AN OPTIONAL ARGUMENT A LIST OF VERTICES OVER WHICH TO APPLY A SZ. THE RESULTANT NETWORK CAN THEN BE CONTRACTED AND DIVIDED BY THE ACTUAL PARTITION FUNCTION TO GET THAT OBSERVABLE
-INDSNETWORK IS ASSUMED TO BE BUILT FROM A GRAPH (NO SITE INDS) AND OF LINK SPACE 2
+OPTIONAL ARGUMENT:
+  h: EXTERNAL MAGNETIC FIELD
+  szverts: A LIST OF VERTICES OVER WHICH TO APPLY A SZ.
+    THE RESULTANT NETWORK CAN THEN BE CONTRACTED AND DIVIDED BY THE ACTUAL PARTITION FUNCTION TO GET THAT OBSERVABLE
+    INDSNETWORK IS ASSUMED TO BE BUILT FROM A GRAPH (NO SITE INDS) AND OF LINK SPACE 2
 """
-function ising_network(eltype::Type, s::IndsNetwork, beta::Number; szverts=nothing)
-  ψ = delta_network(eltype, s)
-
+function ising_network(
+  eltype::Type, s::IndsNetwork, beta::Number; h::Number=0.0, szverts=nothing
+)
+  tn = delta_network(eltype, s)
   if (szverts != nothing)
     for v in szverts
-      ψ[v] = diagITensor(eltype[1, -1], inds(ψ[v]))
+      tn[v] = diagITensor(eltype[1, -1], inds(tn[v]))
     end
   end
-  J = 1
-  f1 = 0.5 * sqrt(exp(-J * 2 * beta * 0.5) * (-1 + exp(J * 2 * beta)))
-  f2 = 0.5 * sqrt(exp(-J * 2 * beta * 0.5) * (1 + exp(J * 2 * beta)))
-  q = eltype[(f1+f2) (-f1+f2); (-f1+f2) (f1+f2)]
-  for v in vertices(ψ)
-    is = inds(ψ[v])
-    indices = inds(ψ[v])
-    for i in indices
-      qtens = itensor(q, i, i')
-      ψ[v] = ψ[v] * qtens
-      ψ[v] = noprime!(ψ[v])
-    end
+  for edge in edges(tn)
+    v1 = src(edge)
+    v2 = dst(edge)
+    i = commoninds(tn[v1], tn[v2])[1]
+    deg_v1 = degree(tn, v1)
+    deg_v2 = degree(tn, v2)
+    f11 = exp(beta * (1 + h / deg_v1 + h / deg_v2))
+    f12 = exp(beta * (-1 + h / deg_v1 - h / deg_v2))
+    f21 = exp(beta * (-1 - h / deg_v1 + h / deg_v2))
+    f22 = exp(beta * (1 - h / deg_v1 - h / deg_v2))
+    q = eltype[f11 f12; f21 f22]
+    w, V = eigen(q)
+    w = map(sqrt, w)
+    sqrt_q = V * LinearAlgebra.diagm(w) * inv(V)
```

Comment:
You should be able to use `Diagonal(w)` here.

Reply:
SG, done!